### DISCUSSION TITLE
Testing lazy props

### DISCUSSION THREAD

**Discussion by captenmasin** [inertiajs/inertia#604] [Help]
Using the new test helpers

```
$this->get('/')->assertInertia(function (Assert $inertia) {
        $inertia->component('Dashboard')->has('user');
    });
```

How would one assert that a prop exists when it's loaded via `Inertia::lazy()` ?

**Comment by reinink** [comment]
Hey @claudiodekker, I know we've been asked about this before. Do we have a good solution for this yet? 🤔

**Reply by claudiodekker** [reply]
Not yet! I think we'll have to end up making a custom Trait that allows you to do things like that. 
Then you'd just use that in your `TestCase` & use something like `inertia()->get()` / `inertiaGet()` or something.

**Reply by jackkitley** [reply]
still nothing on this?

**Reply by JeroenGovers** [reply]
I ran into the same problem. So I created a pull request [#595](https://github.com/inertiajs/inertia-laravel/pull/595)

**Comment by lukasleitsch** [comment]
This way works for me:

```php
$this
    ->withHeaders([
        'X-Inertia-Partial-Component' => 'Home/Home',
        'X-Inertia-Partial-Data' => 'users',
    ])
    ->get('/')
    ->assertOK()
    ->assertInertia(fn(AssertableInertia $page) => $page
        ->has('users')
        ->etc()
    );
```

Credits: https://twitter.com/djgeisi/status/1637803207324778498

**Reply by Barbapapazes** [reply]
While this is a great solution, this requires us to know how inertia works under the hood. They could expose the headers through constants to make it easier (and maybe explain them in the documentation).

**Comment by PerryRylance** [comment]
Just arrived here looking for how to test lazy props - would be great if we could have `withLazyProps`, `withOnly` or something

**Comment by IronSinew** [comment]
Bubbling this back to the top, is this something that is intended to have official support eventually?
